pkg/lint/fileparser.go

```diff
@@ -0,0 +1,287 @@
+package lint
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"regexp"
+	"strconv"
+	"strings"
+	"text/template/parse"
+
+	"github.com/Masterminds/sprig"
+	"github.com/ghodss/yaml"
+
+	"github.com/giantswarm/microerror"
+	pathmodifier "github.com/giantswarm/valuemodifier/path"
+)
+
+var (
+	fMap                 = dummyFuncMap()
+	includes             = &includeExtract{[]string{}}
+	templatePathPattern  = regexp.MustCompile(`(\.[a-zA-Z].[a-zA-Z0-9_\.]+)`)
+	yamlErrorLinePattern = regexp.MustCompile(`yaml: line (\d+)`)
+)
+
+func init() {
+	fMap["include"] = includes.include
+}
+
+type ValueFile struct {
+	filepath     string
+	installation string // optional
+	paths        map[string]*ValuePath
+	sourceBytes  []byte
+}
+
+type ValuePath struct {
+	Value interface{}
+	// files using this value
+	UsedBy []*TemplateFile
+	// value is overshadowed by some files
+	OvershadowedBy []*ValueFile
+}
+
+type TemplateFile struct {
+	filepath     string
+	installation string // optional for defaults
+	app          string
+
+	// values map contains values requested in template using template's dot
+	// notation, e.g. '{{ .some.value }}'
+	values map[string]*TemplateValue
+	// paths map contains all paths in template extracted by valuemodifier/path
+	paths map[string]bool
+	// includes contains names of all include files used by this template
+	includes []string
+
+	sourceBytes    []byte
+	sourceTemplate *template.Template
```

Comment:
I think we can remove `sourceBytes` and `sourceTemplates` everywhere. Are they used?